app/controllers/wpcc/your_details_controller.rb

```diff
@@ -48,6 +51,12 @@ def session_params
 
     def amend_session
       your_details_form_params.each { |key, value| session[key] = value }
+      session[:manually_opt_in] = manually_opt_in?
+    end
+
+    def manually_opt_in?
+      salary = your_details_form_params[:salary].to_i
```

Comment:
We risk losing some values by converting this into an integer. How about,..

`salary = your_details_form_params[:salary].to_i` --> `salary = your_details_form_params[:salary].to_f`